model/src/payout_curve.rs

```diff
@@ -820,6 +821,83 @@ mod tests {
         pretty_assertions::assert_eq!(actual_payouts.last().unwrap(), &upper_tail);
     }
 
+
+    proptest! {
+        /// By similar we mean that they're at most 1 satoshi off the
+        /// next payout sum.
+        #[test]
+        fn payout_totals_are_similar(
+            price in arb_price(1.0, 340_000.0),
+            n_contracts in arb_contracts(1, 10_000_000),
+            taker_leverage in arb_leverage(1, 200),
+            n_payouts in 10usize..2_000,
+            fee_flow in arb_fee_flow(0, 100_000_000),
+        ) {
+            let payouts = calculate_payout_parameters(
+                price,
+                n_contracts,
+                taker_leverage,
+                n_payouts,
+                fee_flow,
+            )
+            .unwrap();
+
+            let are_payout_totals_similar = payouts
+                .iter()
+                .zip(payouts.iter().skip(1))
+                .map(|(a, b)| {
+                    (
+                        (a.long_amount + a.short_amount),
+                        (b.long_amount + b.short_amount),
+                    )
+                })
+                .all(|(a, b)| (a as i64 - b as i64).abs() <= 1);
+
+            prop_assert!(are_payout_totals_similar)
```

Comment:
So with this you are asserting that given a window of two ranges, the payouts are the same?

Reply:
Basically that contiguous payouts are close. Obviously not what I originally intended because I wanted to use `all_equal`. How would you define a `all_almost_equal` combinator?

Reply:
The definition looks good to me, I was just surprised that we want to test this property here!

Reply:
I suppose we don't need an assertion. Originally I set out to verify that we were not panicking or failing when generating the payouts.

This assertion proved useful though, since one of the error cases saw us overflowing and generating some payouts with insanely high values.

Reply:
Another one could be that we are always outputting the same sum of sats.

Reply:
> Another one could be that we are always outputting the same sum of sats.

I think that's what I was trying to approximate with the assertion above.